lib/youtube-dl.js

```diff
@@ -256,12 +262,34 @@ ytdl.getFormats = function(url, args, callback) {
       var result = formatsRegex.exec(line);
       if (/\[info\]/.test(line)) { status = line.split(' ')[4].slice(0, -1); }
       if (result) {
-        formats.push({
+        var optionalResult = optionalSpecsRegex.exec(result[4])
+        var compiledResult = {
           id         : status,
-          itag       : parseInt(result[1], 10),
+          code       : result[1],
           filetype   : result[2],
           resolution : result[3],
-        });
+        }
+        if (/^\d+$/.test(compiledResult.code)) {
```

Comment:
The `itag` attributes will be set for any video with an integer `code` value. So the `itag` would be set for youtube and any of the other ~500+ web sites supported by youtube dl in theory that use integer `code` values (if any).